### PR TITLE
Refactor Plausible.Billing.Plans module

### PR DESCRIPTION
**Pull request by vinibrsl**
This commit introduces a series of improvements on the Plans module including function renaming, documentation and readability. This is the groundwork for billing plans.

There should be no _actual_ changes with this commit, therefore no changes in tests either.

### CODE REVIEW

File: lib/plausible/billing/plans.ex [hunk 1]
```diff
@@ -1,13 +1,18 @@
 defmodule Plausible.Billing.Plans do
   use Plausible.Repo
 
-  @unlisted_plans_v1 [
-    %{limit: 150_000_000, yearly_product_id: "648089", yearly_cost: "$4800"}
-  ]
+  for f <- [:plans_v1, :plans_v2, :plans_v3] do
```
Comment (aerosol) [inline lib/plausible/billing/plans.ex:4]: Have you decided not to use @external_resource annotation?

**Review by aerosol — APPROVED** [review]
LGTM. I would include the `@external_resource` annotation, unless we have reasons not to.

File: lib/plausible/billing/plans.ex [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+    contents =
+      :plausible
+      |> Application.app_dir(["priv", "#{f}.json"])
+      |> File.read!()
+      |> Jason.decode!(keys: :atoms)
```
[the remark this answers is not in the document]
Comment (vinibrsl) [reply lib/plausible/billing/plans.ex]: What do you think of 6c610009cb2d0acc5d2656ca4928cfa7c9f31dc6?

[the remark this answers is not in the document]
Comment (aerosol) [reply lib/plausible/billing/plans.ex]: 😻
😻
😻

**Pull request merged**